flocker/control/schema/endpoints.yml

```diff
@@ -12,3 +12,61 @@ definitions:
     required:
       - flocker
     additionalProperties: false
+
+  datasets:
+    type: object
+    properties:
+      dataset_id:
+        title: "Unique identifier"
+        description: |
+          An opaque identifier, unique across the cluster, identifying a
+          particular dataset.  If not given, a new identifier will be generated
+          and returned.
+        type: string
+        # The length of a stringified uuid
+        minLength: 36
+        maxLength: 36
+
+      metadata:
+        title: "Data about a dataset"
+        description: |
+          Additional key/value data describing the dataset.  These items are
+          not interpreted by Flocker.  If not given, no metadata will be
+          associated with the new dataset.
+        type: object
+        patternProperties:
+          ".{1,256}":
+            type: string
+            # Arbitrary
```

Comment:
:question: I think this and the following comment should be combined and expanded. Something like "we limit the length/number of properties for this and this reason. The specific limits are arbitrary."